tests/xharness/AppRunner.cs

```diff
@@ -332,6 +420,11 @@ public int Install (Log log)
 			args.Append (" -setenv=NUNIT_AUTOEXIT=true");
 			args.Append (" -argument=-app-arg:-enablenetwork");
 			args.Append (" -setenv=NUNIT_ENABLE_NETWORK=true");
+			// detect if we are using a jenkins bot.
+			if (!string.IsNullOrEmpty (Environment.GetEnvironmentVariable ("BUILD_REVISION"))
```

Comment:
There's already a Harness.InWrench method, so it makes sense to extract this to a Harness.InJenkins method as well.